sound/soc/sof/intel/hda.c

```diff
@@ -1117,14 +1119,145 @@ static void hda_generic_machine_select(struct snd_sof_dev *sdev,
 
 #if IS_ENABLED(CONFIG_SND_SOC_SOF_INTEL_SOUNDWIRE)
 
+static bool is_endpoint_present(struct sdw_slave *sdw_device,
+				struct asoc_sdw_codec_info *dai_info, int dai_type)
+{
+	int i;
+
+	for (i = 0; i < sdw_device->sdca_data.num_functions; i++) {
+		if (dai_type == dai_info->dais[i].dai_type)
+			return true;
+	}
+	dev_dbg(&sdw_device->dev, "Endpoint DAI type %d not found\n", dai_type);
```

Comment:
this is not an error, right?

do we need to print this and would we want to print if it is found also?

Reply:
No, it is not an error. I only print when the dai_type is not fond because that the dai_type will be found in most cases. And that is expected. Printing it makes users know exactly the reason why the endpoint doesn't show up.